inc/internal/az_ulib_ipc.h

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license.
+// See LICENSE file in the project root for full license information.
+
+#ifndef INTERNAL_AZ_ULIB_IPC_H
+#define INTERNAL_AZ_ULIB_IPC_H
+
+#include "azure_macro_utils/macro_utils.h"
+#include "umock_c/umock_c_prod.h"
+
+#include "../az_ulib_action_api.h"
+#include "../az_ulib_descriptor_api.h"
+#include "az_pal_os_api.h"
+#include "az_ulib_action_api.h"
+#include "az_ulib_base.h"
+#include "ulib_config.h"
+#include "ulib_port.h"
+#include "ulib_result.h"
+
+#ifndef __cplusplus
+#include <stdint.h>
+#else
+#include <cstdint>
+extern "C" {
+#endif
+
+typedef void *_az_ulib_ipc_interface_handle;
+
+typedef struct _az_ulib_ipc_interface_tag {
+  const az_ulib_interface_descriptor *interface_descriptor;
+  volatile long ref_count;
+  volatile long running_count;
+} _az_ulib_ipc_interface;
+
+typedef struct _az_ulib_ipc_tag {
+  AZ_PAL_OS_LOCK lock;
+  _az_ulib_ipc_interface interface_list[AZ_ULIB_CONFIG_MAX_IPC_INTERFACE];
+} _az_ulib_ipc;
+
+MOCKABLE_FUNCTION(, AZ_ULIB_RESULT, _az_ulib_ipc_init_no_test, _az_ulib_ipc *, ipc_handle);
+MOCKABLE_FUNCTION(, AZ_ULIB_RESULT, _az_ulib_ipc_init, _az_ulib_ipc *, ipc_handle);
+
+MOCKABLE_FUNCTION(, AZ_ULIB_RESULT, _az_ulib_ipc_deinit_no_test);
+MOCKABLE_FUNCTION(, AZ_ULIB_RESULT, _az_ulib_ipc_deinit);
+
+MOCKABLE_FUNCTION(
+    ,
+    AZ_ULIB_RESULT,
+    _az_ulib_ipc_publish_no_test,
+    const az_ulib_interface_descriptor *,
+    interface_descriptor);
+MOCKABLE_FUNCTION(
+    ,
+    AZ_ULIB_RESULT,
+    _az_ulib_ipc_publish,
+    const az_ulib_interface_descriptor *,
+    interface_descriptor);
+
+MOCKABLE_FUNCTION(
+    ,
+    AZ_ULIB_RESULT,
+    _az_ulib_ipc_unpublish_no_test,
+    const az_ulib_interface_descriptor *,
+    interface_descriptor,
+    uint32_t,
+    wait_option_ms);
+MOCKABLE_FUNCTION(
```

Comment:
Nit: make spacing consistent (i.e. either a line separation for each of them or none), but I would vote for a line separation on each.